### PR TITLE
add Golden Candy Corn

### PR DESCRIPTION
**Pull request by LemmaEOF**
- Golden Candy Corn can be gotten from trick or treating (weight of 1), spider lair chests (1% chance), or killing witches or pumpcowns (1% chance each).
- Golden Candy Corn restores 1 hunger and 25% saturation, but each item can be eaten 250 times (not stackable due to durability).
- Once you've started eating Golden Candy Corn, you will continue to eat it again every 2 seconds until you either remove it from your main or off hand, or your hunger fills up.
- Added a `SpookyItems.newSettings()` method to create new `Item.Settings` objects set to the spookytime creative tab.

### CODE REVIEW

**Review by Shnupbups — APPROVED** [review]

File: src/main/java/com/fabriccommunity/spookytime/item/GoldenCandyCornItem.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+public class GoldenCandyCornItem extends CandyItem {
+	public GoldenCandyCornItem(Settings settings, int hunger, float saturation) {
+		super(settings, hunger, saturation);
+	}
+
+	@Override
+	public ItemStack finishUsing(ItemStack stack, World world, LivingEntity user) {
+		if (user instanceof PlayerEntity) {
+			PlayerEntity player = (PlayerEntity)user;
```
Comment (emilyploszaj) [inline src/main/java/com/fabriccommunity/spookytime/item/GoldenCandyCornItem.java]: Space after the cast here.

File: src/main/java/com/fabriccommunity/spookytime/mixin/LivingEntityMixin.java [hunk 2]
```diff
@@ -14,6 +17,7 @@
 
 import com.fabriccommunity.spookytime.enchantment.BeheadingEnchantment;
 import com.fabriccommunity.spookytime.enchantment.LifestealEnchantment;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
 
 /**
  * Implement Beheading and Lifesteal.
```
Comment (emilyploszaj) [inline src/main/java/com/fabriccommunity/spookytime/mixin/LivingEntityMixin.java]: Update description of what this mixin does.

**Review by emilyploszaj — COMMENTED** [review]
🦀 Style police is here 🦀

**Review by emilyploszaj — APPROVED** [review]

**Review by Shrimpstress — APPROVED** [review]
Looked thoroughly, everything looks good to me.

**Review by Draylar — APPROVED** [review]

**Review by Juuxel — APPROVED** [review]

**Pull request merged**